src/com/facebook/buck/android/AndroidCoreArg.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2015-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.facebook.buck.android;
+
+import com.google.common.collect.ImmutableList;
+import java.util.Optional;
+
+public interface AndroidCoreArg {
```

Comment:
Last small nit: these fields don't seem related to "AndroidCore". `AndroidLibraryDescription.CoreArg` is more "core" than this. Maybe rename to AndroidKotlinCoreArg?